x-pack/plugins/enterprise_search/public/applications/app_search/components/api_logs/api_logs.test.tsx

```diff
@@ -42,19 +39,28 @@ describe('ApiLogs', () => {
 
   it('renders', () => {
     const wrapper = shallow(<ApiLogs />);
-    expect(wrapper.find(EuiPageHeader).prop('pageTitle')).toEqual('API Logs');
+    expect(getPageTitle(wrapper)).toEqual('API Logs');
     expect(wrapper.find(ApiLogsTable)).toHaveLength(1);
     expect(wrapper.find(NewApiEventsPrompt)).toHaveLength(1);
 
     expect(wrapper.find(LogRetentionCallout).prop('type')).toEqual('api');
     expect(wrapper.find(LogRetentionTooltip).prop('type')).toEqual('api');
   });
 
-  it('renders a loading screen', () => {
-    setMockValues({ ...values, dataLoading: true, apiLogs: [] });
-    const wrapper = shallow(<ApiLogs />);
+  describe('loading state', () => {
+    it('renders a full-page loading state on initial page load (no logs exist yet)', () => {
+      setMockValues({ ...values, dataLoading: true, apiLogs: [] });
+      const wrapper = shallow(<ApiLogs />);
+
+      expect(wrapper.prop('isLoading')).toEqual(true);
+    });
+
+    it('does not re-render a full-page loading state after initial page load (uses component-level loading state instead)', () => {
```

Comment:
Note that the component level loading here is the EuiBasicTable's default `loading` prop. The only time data reloads on this page is when users are paginating between pages of logs, so it makes sense to handle that loading UX at the component level instead of re-flashing a full-page loading state.

The test name was already getting really long though so I didn't want to add all that extra context to it 🤪